src/Symfony/Component/HttpKernel/HttpCache/HttpCache.php

```diff
@@ -374,7 +374,11 @@ protected function validate(Request $request, Response $entry, $catch = false)
         $subRequest = clone $request;
 
         // send no head requests because we want content
-        $subRequest->setMethod('GET');
+        if ($request->isMethodSafe() && $request->getMethod() !== 'HEAD') {
+            $subRequest->setMethod($request->getMethod());
```

Comment:
Is this needed. By default, the subrequest is going to have the same method as the main one.